c_src/fast_yaml.c

```diff
@@ -100,16 +119,41 @@ static int make_atom(ErlNifEnv* env, yaml_event_t *event)
   }
 }
 
-static ERL_NIF_TERM make_scalar(ErlNifEnv* env, yaml_event_t *event, int flags)
+static ERL_NIF_TERM make_scalar(ErlNifEnv* env, yaml_event_t *event, int flags, int is_map_value)
 {
     int as_atom = PLAIN_AS_ATOM & flags;
+    int sane_scalars = SANE_SCALARS & flags;
+
     long int i;
     double d;
     int type;
     yaml_scalar_style_t style = event->data.scalar.style;
     ERL_NIF_TERM rterm;
 
-    if (as_atom && style == YAML_SINGLE_QUOTED_SCALAR_STYLE) {
+    if (sane_scalars) {
+        if (is_map_value && style == YAML_PLAIN_SCALAR_STYLE) {
+            if ((type = make_num(env, event->data.scalar.value,
```

Comment:
Do you think you could refactor make_num to return erlang term directly? Just pretty much inline that type check inside that function, you could drop those pointers to int and double, and that function already takes env so it can create terms. There is another place when make_num is used, so that would cut on duplicate code.

Reply:
Okay, I refactored it. Code was originally by @zinid.